## Observations

<b>Positive</b>
- Data types of all the columns seems correct except "Sent_date" which is string
- All the flag columns values are either 1 or 0
- is_sent column value is always 1
- 


<b>Negative</b>
- Had issue while reading the csv file. Encoding other than utf-8 passed
- All the columns except "Primary Topic" are not not null. (Find the reason why primary topic is NULL)
- Understand more about email campign id as it is not PK.
- "Primary Topic" Column has special characters that doesn't mean anything (/, --). Spelling errors

In [1]:
import pandas as pd

In [5]:
email_path = "../data/Email_Campaign_Details.csv"
# email_df = pd.read_csv(email_path, encoding='cp1252')
email_df = pd.read_csv(email_path, encoding='mac_roman')
email_df.head()

Email_Campaign_id          contact_id               email_subject  \
0  701200000006cIGAAY  00v2000000IZVYWAA5  Roadshow Houston (Advisor)   
1  701200000006cIGAAY  00v2000000IZW13AAH  Roadshow Houston (Advisor)   
2  701200000006cIGAAY  00v2000000IZUepAAH  Roadshow Houston (Advisor)   
3  701200000006cIGAAY  00v2000000IZWDdAAP  Roadshow Houston (Advisor)   
4  701200000006cIGAAY  00v2000000IZUdSAAX  Roadshow Houston (Advisor)   

   is_sent  is_delivered  is_link_clicked  is_hard_bounce  is_bounce  \
0        1             1                0               0          0   
1        1             1                0               0          0   
2        1             1                0               0          0   
3        1             1                0               0          0   
4        1             1                0               0          0   

   is_opt_out  is_open Sent_date       Campaign_type Primary Topic  
0           0        1  2/7/2023  Events - In Person             /  
1           0        1  2/7/2023  Events - In Person             /  
2           0        0  2/7/2023  Events - In Person             /  
3           0        0  2/7/2023  Events - In Person             /  
4           0        0  2/7/2023  Events - In Person             /

In [6]:
email_df.dtypes

Email_Campaign_id    object
contact_id           object
email_subject        object
is_sent               int64
is_delivered          int64
is_link_clicked       int64
is_hard_bounce        int64
is_bounce             int64
is_opt_out            int64
is_open               int64
Sent_date            object
Campaign_type        object
Primary Topic        object
dtype: object

In [7]:
email_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 128236 entries, 0 to 128235
Data columns (total 13 columns):
 #   Column             Non-Null Count   Dtype 
---  ------             --------------   ----- 
 0   Email_Campaign_id  128236 non-null  object
 1   contact_id         128236 non-null  object
 2   email_subject      128236 non-null  object
 3   is_sent            128236 non-null  int64 
 4   is_delivered       128236 non-null  int64 
 5   is_link_clicked    128236 non-null  int64 
 6   is_hard_bounce     128236 non-null  int64 
 7   is_bounce          128236 non-null  int64 
 8   is_opt_out         128236 non-null  int64 
 9   is_open            128236 non-null  int64 
 10  Sent_date          128236 non-null  object
 11  Campaign_type      128236 non-null  object
 12  Primary Topic      111816 non-null  object
dtypes: int64(7), object(6)
memory usage: 12.7+ MB


In [10]:
email_df.describe(include = 'all')

Email_Campaign_id          contact_id email_subject   is_sent  \
count               128236              128236        128236  128236.0   
unique                 514                3872           466       NaN   
top     70120000000MfxSAAS  00v2000000IZVYWAA5  ETF Review 8       NaN   
freq                  3778                  35          3778       NaN   
mean                   NaN                 NaN           NaN       1.0   
std                    NaN                 NaN           NaN       0.0   
min                    NaN                 NaN           NaN       1.0   
25%                    NaN                 NaN           NaN       1.0   
50%                    NaN                 NaN           NaN       1.0   
75%                    NaN                 NaN           NaN       1.0   
max                    NaN                 NaN           NaN       1.0   

         is_delivered  is_link_clicked  is_hard_bounce      is_bounce  \
count   128236.000000    128236.000000   128236.000000  128236.000000   
unique            NaN              NaN             NaN            NaN   
top               NaN              NaN             NaN            NaN   
freq              NaN              NaN             NaN            NaN   
mean         0.978243         0.004616        0.000686       0.022240   
std          0.145889         0.067788        0.026187       0.147465   
min          0.000000         0.000000        0.000000       0.000000   
25%          1.000000         0.000000        0.000000       0.000000   
50%          1.000000         0.000000        0.000000       0.000000   
75%          1.000000         0.000000        0.000000       0.000000   
max          1.000000         1.000000        1.000000       1.000000   

           is_opt_out        is_open Sent_date Campaign_type Primary Topic  
count   128236.000000  128236.000000    128236        128236        111816  
unique            NaN            NaN       219             6            18  
top               NaN            NaN  6/7/2023    Newsletter           ETF  
freq              NaN            NaN      6102         39060         34706  
mean         0.000094       0.260036       NaN           NaN           NaN  
std          0.009673       0.438656       NaN           NaN           NaN  
min          0.000000       0.000000       NaN           NaN           NaN  
25%          0.000000       0.000000       NaN           NaN           NaN  
50%          0.000000       0.000000       NaN           NaN           NaN  
75%          0.000000       1.000000       NaN           NaN           NaN  
max          1.000000       1.000000       NaN           NaN           NaN

There are only 514 unique email campaign ids. Need to understand more about email campaign id



In [12]:
email_df["Campaign_type"].value_counts()

Campaign_type
Newsletter                 39060
Marketing Campaign         32026
Event follow up            22497
Digital event              16538
Events - In Person         10024
Marketing Communication     8091
Name: count, dtype: int64

In [13]:
email_df["Primary Topic"].value_counts()

Primary Topic
ETF                34706
Bonds              14321
Portfolio          13840
Market Insights    13226
Ad hoc              8632
Adhoc               7167
Equities            6514
Insights            5326
HealthCare          4943
Asssets             1095
Funds                814
Income               465
Investment           449
Retirement           187
Asset                 61
--                    39
/                     25
Growth                 6
Name: count, dtype: int64

In [14]:
email_df.is_sent.value_counts()

is_sent
1    128236
Name: count, dtype: int64

In [16]:
email_df.is_delivered.value_counts()

is_delivered
1    125446
0      2790
Name: count, dtype: int64